crates/typst-syntax/src/parser.rs

```diff
@@ -47,15 +47,11 @@ fn markup_exprs(p: &mut Parser, mut at_start: bool, stop_set: SyntaxSet) {
     debug_assert!(stop_set.contains(SyntaxKind::End));
     at_start |= p.had_newline();
     let mut nesting: usize = 0;
-    loop {
-        match p.current() {
-            SyntaxKind::LeftBracket => nesting += 1,
-            SyntaxKind::RightBracket if nesting > 0 => nesting -= 1,
-            _ if p.at_set(stop_set) => break,
-            _ => {}
-        }
-        markup_expr(p, at_start);
+    let mut at_nested_bracket = false;
+    while !p.at_set(stop_set) || at_nested_bracket {
+        markup_expr(p, at_start, &mut nesting);
         at_start = p.had_newline();
+        at_nested_bracket = nesting > 0 && p.at(SyntaxKind::RightBracket);
```

Comment:
couldn't we remove the variable and directly inline this expression after the `||` in the while loop's condition?

Reply:
Yes we can :+1: